lib/Db/FaceMapper.php

```diff
@@ -131,6 +131,51 @@ public function findFacesFromPerson(string $userId, int $personId, int $model, $
 		return $faces;
 	}
 
+	public function findRepresentativeFromPerson(string $userId, int $personId, float $sensitivity, int $model) {
+		if (!version_compare(phpversion('pdlib'), '1.0.2', '>=')) {
+			return null;
+		}
+		$qb = $this->db->getQueryBuilder();
+		$qb->select('f.id', 'f.image', 'f.person', 'f.descriptor')
+			->from($this->getTableName(), 'f')
+			->innerJoin('f', 'facerecog_images' ,'i', $qb->expr()->eq('f.image', 'i.id'))
+			->where($qb->expr()->eq('user', $qb->createNamedParameter($userId)))
+			->andWhere($qb->expr()->eq('person', $qb->createNamedParameter($personId)))
+			->andWhere($qb->expr()->eq('model', $qb->createNamedParameter($model)));
+		$faces = $this->findEntities($qb);
+
+		$facesCount = array();
+		for ($i = 0, $face_count1 = count($faces); $i < $face_count1; $i++) {
+			$face1 = $faces[$i];
+			for ($j = $i, $face_count2 = count($faces); $j < $face_count2; $j++) {
```

Comment:
you can start with `$i+1` (second loop), as everyone will have `$distance=0` with itself?

Reply:
If I do this, I can't compare the singles clusters..